examples/mnasnet/mnasnet.py

```diff
@@ -0,0 +1,324 @@
+        self.val_set = vision.datasets.CIFAR10(root='./data', train=False,
+                            download=True, transform=self.transform)
+    
+    def _init_test_set(self):
+        self.test_set = vision.datasets.CIFAR10(root='./data', train=False,
+                            download=True, transform=self.transform)
+
+    def train_dataloader(self):
+        kwargs = dict(
+            batch_size=self.batch_size,
+            shuffle=True,
+            num_workers=2
+        )
+        self.train_loader = torch.utils.data.DataLoader(self.train_set, **kwargs)
+        return self.train_loader
+    
+    def val_dataloader(self):
+        kwargs = dict(
+            batch_size=self.batch_size,
+            shuffle=True,
+            num_workers=2
+        )
+        self.val_loader = torch.utils.data.DataLoader(self.val_set, **kwargs)
+        return self.val_loader
+    
+    def test_dataloader(self, batch_size):
+        kwargs = dict(
+            batch_size=batch_size,
+            shuffle=True,
+            num_workers=2
+        )
+        self.test_loader = torch.utils.data.DataLoader(self.val_set, **kwargs)
+        return self.test_loader
+
+@METRICS.register()
+class MyClassificationMetrics(pl.metrics.Metric):
+    def __init__(self, num_classes, *, acc={}, f1={}, prec={}, rec={}):
+        super().__init__()
+        prec.update(dict(num_classes=num_classes))
+        rec.update(dict(num_classes=num_classes))
+        f1.update(dict(num_classes=num_classes))
+        self.metrics = nn.ModuleList([
+            pl.metrics.Accuracy(**acc),
+            pl.metrics.F1(**f1),
+            pl.metrics.Precision(**prec),
+            pl.metrics.Recall(**rec),
+        ])
+        self.metrics_args = dict(
+            num_classes=num_classes,
+            acc=acc, f1=f1, prec=prec, rec=rec
+        )
+
+    def update(self, inputs, targets):
+        if isinstance(inputs, list):
+            inputs = np.concatenate([inp['class_label'] for inp in inputs])
+            inputs = torch.as_tensor(inputs).to(targets.device)
+        for metric in self.metrics:
+            metric.update(inputs, targets)
+    
+    def compute(self):
+        results = {}
+        typename = lambda x: type(x).__name__.split('.')[-1]
+        for metric in self.metrics:
+            name = typename(metric)
+            results[name] = metric.compute()
+        return results
+
+# %%
+# 2. Preparing the Model
+# ----------------------
+# `vortex`'s `ModelBase` is just an extension to `pl.LightningModule`,
+# so we can use it just like any `LightningModule`, like define
+# `training_step`, `validation_step`, `configure_optimizers`, etc.
+# 
+# Additionally, we need to define the following methods:
+# - `input_names` should return a list of string representing the input names
+# - `output_names` should return a list of string representing the output names
+# - `on_export_start` (optional): will be called by vortex onnx exporter at the start of export
```

Comment:
there is `on_export_end` as well right?

Reply:
well, it is not used in this example

Reply:
ahh I see, probably better mentioned it here as well (?)

Reply:
I don't know, in my opinion it is better to not overload the reader with to much info. Also, it is already documented in ModelBase API reference anyway

Reply:
**too much

Reply:
yepp, I understand